app/inpututils.cpp

```diff
@@ -1354,15 +1381,65 @@ QString InputUtils::distanceToFeature( QgsPoint gpsPosition, const FeatureLayerP
     catch ( QgsCsException &e )
     {
       Q_UNUSED( e )
-      return QString();
+      return -1;
     }
   }
 
   QgsDistanceArea distanceArea;
   distanceArea.setSourceCrs( mapSettings->destinationCrs(), mapSettings->transformContext() );
-  double dist = distanceArea.measureLine( gpsPosition, targetPoint );
-  QString res = formatDistance( dist, distanceArea.lengthUnits(), 2 );
-  return res;
+  qreal distance = distanceArea.measureLine( gpsPosition, targetPoint );
+  distance = distanceArea.convertLengthMeasurement( distance, QgsUnitTypes::DistanceMeters );
+  return distance;
+}
+
+qreal InputUtils::angleBetweenGpsAndFeature( QgsPoint gpsPoint, const FeatureLayerPair &targetFeature, QgsQuickMapSettings *mapSettings )
+{
+  if ( !mapSettings || !targetFeature.isValid() )
+    return -1;
+
+  QgsVectorLayer *layer = targetFeature.layer();
+  QgsFeature f = targetFeature.feature();
+
+  // Only points are supported
+  if ( layer->geometryType() != QgsWkbTypes::GeometryType::PointGeometry )
+    return -1;
+
+  // Transform gps position to map CRS
+  QgsCoordinateReferenceSystem wgs84( "EPSG:4326" );
```

Comment:
why this differs from the line (686) in previous function?
QgsCoordinateReferenceSystem crsGPS = coordinateReferenceSystemFromEpsgId( 4326 )

Reply:
Unified now